File: go/the_highest_profit_wins.go
// https://www.codewars.com/kata/559590633066759614000063

package kata

func MinMax(arr []int) [2]int {
	return [2]int{min(arr), max(arr)}
}

func min(arr []int) int {
	min := arr[0]
	for _, i := range arr {
		if i < min {
			min = i
		}
	}
	return min
}

func max(arr []int) int {
	max := arr[0]
	for _, i := range arr {
		if i > max {
			max = i
		}
	}
	return max
}

File: rust/the_highest_profit_wins.rs
//https://www.codewars.com/kata/559590633066759614000063
fn min_max(lst: &[i32]) -> (i32, i32) {
    (min(lst), max(lst))
}

fn min(arr: &[i32]) -> i32 {
    let mut min = arr[0];
    for i in arr {
        if *i < min {
            min = *i;
        }
    }
    min
}
fn max(arr: &[i32]) -> i32 {
    let mut max = arr[0];
    for i in arr {
        if *i > max {
            max = *i;
        }
    }
    max
}